tubular/misc.py

```diff
@@ -45,3 +46,43 @@ def transform(self, X):
         X[self.columns] = self.value
 
         return X
+
+
+class SetColumnDtype(BaseTransformer):
+    """
+    Transformer to set transform columns in a dataframe to a dtype
+
+    Parameters
+    ----------
+    columns : str or list
+        Columns to set dtype. Must be set or transform will not run.
+
+    dtype : type
+        Dtype to set columns to
+    """
+
+    def __init__(self, columns, dtype):
+
+        super().__init__(columns, copy=True)
+
+        self.__validate_dtype(dtype)
+
+        self.dtype = dtype
+
+    def transform(self, X):
+
+        X = super().transform(X)
+
+        X[self.columns] = X[self.columns].astype(self.dtype)
+
+        return X
+
+    def __validate_dtype(self, dtype: str):
```

Comment:
The self.classname in the error can be really handy when you have dozens of steps in a pipeline.  Always nice to know where things are breaking.  I agree it is slightly overwrought but since it has been done now I'm inclined to leave it in.